cmd/controller-manager/main.go

```diff
@@ -114,6 +116,11 @@ func main() {
 	if err != nil {
 		glog.Fatalf("failed to get advanced-statefulset Clientset: %v", err)
 	}
+	// TODO: optimize the read of genericCli with the shared cache
+	genericCli, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
```

Comment:
Using the client of `controller-runtime` without using the `informer` of `controller-runtime` has a drawback that all read operations will go to kube-apiserver, it is acceptable now because the only read operation is to get the latest version when perform a `CreateOrUpdate`.